core/src/main/java/io/questdb/cutlass/pgwire/PGConnectionContext.java

```diff
@@ -2063,7 +2067,11 @@ private void prepareForNewBatchQuery() {
             isEmptyQuery = false;
             Misc.clear(bindVariableService);
             currentCursor = Misc.free(currentCursor);
-            typesAndInsert = null;
+            if (typesAndInsert != null && !typesAndInsert.hasBindVariables()) { // not cached, need close
```

Comment:
why we only close now when `!typesAndInsert.hasBindVariables()` ?

Reply:
nit: can you re-prase the comment to something like

```
// Insert plan is cached when typesAndInsert has bind variables only for legacy PG server. Do not close typesAndInsert if it is cached. See processCompiledQuery() method, case CompiledQuery.INSERT_AS_SELECT
```

Also, ideally match the if condition on the referenced line, e.g., on line #2330 use

```
if (typesAndInsert.hasBindVariables()) {
```